content/python/concepts/dunder-methods/terms/new/new.md

```diff
@@ -0,0 +1,86 @@
+---
+Title: '__new__()'
+Description: 'Static method which creates a new instance of a class'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+  - 'Web Development'
+Tags:
+  - 'Attributes'
+  - 'Classes'
+  - 'Methods'
+  - 'Objects'
+  - 'OOP'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+The **`__new__()`** method creates a new instance of a class cls and takes in the class (of which the instance was requested) as the first argument.
+
+## Syntax
+
+```pseudo
+class ClassName:
+  __new__(cls, *args, **kwargs):
+    obj = super().__new__(cls)
+    return obj
+```
+
+Object creation is typically done by invoking the superclass’ `__new__()` in two ways. The first argument of `__new__()` is always the class which is passed in as the first parameter automatically. The name cls is not a keyword; it’s used to reference the class (the first parameter) by convention. When `__new__()` returns an instance of cls it automatically invokes the [`__init__()`](https://www.codecademy.com/resources/docs/python/dunder-methods/init) method of that instance with the arguments passed to it. The first argument passed to `__init__()` will be the instance itself (this happens automatically) which is by convention referenced by self.
+
+In the example above no parent class was specified therefore the class will directly inherit from the object base class (from which all classes inherit). Therefore calling it explicitly using `object.__new__(cls)` will have the same effect.
+
+Let's add the following to the code above:
+
+```pseudo
+def __init__(self, *args, **kwargs):
+    print(*args, **kwargs)
+
+class_instance = ClassName(1, 2)
+```
+First `__new__()` is invoked during the construction of the object (instance of class ClassName) with the arguments 1 and 2 which are accepted within \*args. The object is created by calling the `__new__()` method of the object base class which only takes in the class cls. It then returns an object which is then stored in obj and returned. This invokes the `__init__()` method of the ClassName class with the arguments (obj, \*args, \*\*kwargs).
+
+This method has two main applications:
+
+- Customizing how new class instances are created.
+- Subclassing of immutable types which cannot be initialized via a simple __init__method.
+
+
+## Codebyte Example 1
```

Comment:
We generally only have one codebyte example at the end of the entry.  We can make this just a regular example with a ```py block.